docs/amber_script.md

```diff
@@ -31,11 +31,19 @@ which is the characters `0x` followed by hexadecimal digits.
  * tessellation\_evaluation
  * tessellation\_control
  * compute
+ * multi
 
 The compute pipeline can only contain compute shaders. The graphics pipeline
 can not contain compute shaders, and must contain a vertex shader and a fragment
 shader.
 
+The provided `multi` shader can only be used with `SPIRV-ASM` and `SPIRV-HEX`
+and allows for providing multiple shaders in a single module (so the `vertex`
+and `fragment` shaders can be provided together.)
+
+Note, `SPIRV-ASM` and `SPIRV-HEX` can also be used with each of the other shader
```

Comment:
I don't like this rule.  We often have many compute shaders in the same module which are then distinguished by entry point name.

Reply:
This rule doesn't preclude that? You can have a multi shader with a compute, vertex and fragment. You can't can't have a vertex and compute if it isn't multi. But, you can have a compute SPIRV-ASM as well.

Reply:
Ok.  This works.  It's different from my mental model but it resolves to correct and unambiguous behaviour.